aten/src/ATen/core/ivalue_inl.h

```diff
@@ -479,6 +482,17 @@ struct C10_EXPORT ivalue::Future : c10::intrusive_ptr_target {
     return type_;
   }
 
+ protected:
+  virtual c10::intrusive_ptr<Future> createInstance(at::TypePtr type) {
+    return c10::make_intrusive<Future>(type);
+  }
+
+  virtual void postMarkCompletedHook(const at::IValue& value) {}
+
+  virtual std::function<void(void)> wrapCallback(std::function<void(void)> callback) { return callback; }
+
+  virtual void postWaitHook() {}
```

Comment:
Shall we add some comments to these functions and explain what derived classes need to do when implementing them?